git/repo/base.py

```diff
@@ -713,11 +713,14 @@ def blame_incremental(self, rev, file, **kwargs):
                            committed_date=int(props[b'committer-time']))
                 commits[hexsha] = c
             else:
-                # Discard the next line (it's a filename end tag)
-                line = next(stream)
-                tag, value = line.split(b' ', 1)
-                assert tag == b'filename', 'Unexpected git blame output'
-                orig_filename = value
+                # Discard all lines until we find "filename" which is
+                # guaranteed to be the last line
+                while True:
```

Comment:
I believe this loops break-condition is not clearly defined just now.
Do you think 'end-of-stream' could be another one?

What do you think of something like this:
```python
for line in stream:
	tag, value = line.split(b' ', 1)
	if tag == b'filename':
		orig_filename = value
		break
		
# Finally handle the case of the stream being exhausted too early
```